tuf/api/metadata.py

```diff
@@ -1219,6 +1219,84 @@ def to_dict(self) -> Dict[str, Any]:
             **self.unrecognized_fields,
         }
 
+    @classmethod
+    def from_file(
+        cls, filepath: str, hash_algorithms: Optional[List[str]] = None
+    ) -> "TargetFile":
+        """Creates TargetFile object from a file.
+        Attributes:
+            filepath: The path to the file to create the TaretFile object from.
+            hash_algorithms: An optional list of hash algorithms to create
+                the hashes with. If not specified the securesystemslib default
+                hash algorithm is used.
+        Raises:
+            FileNotFoundError: The file doesn't exist.
```

Comment:
* Arguments, not Attributes
* also raises UnsupportedAlgorithmError

Reply:
@jku, thanks for the feedback and the patience! I've addressed your comments.